pkg/virt-controller/watch/vmi.go

```diff
@@ -320,6 +325,8 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			vmiCopy.Status.Phase = virtv1.Failed
 		} else if hasFailedDataVolume {
 			vmiCopy.Status.Phase = virtv1.Failed
+		} else if syncErr != nil && syncErr.Reason() == FailedPVCVolumeSourceMisusedReason {
+			vmiCopy.Status.Phase = virtv1.Failed
```

Comment:
does it make sense to account for the fact that the vm controller will continually attempt to spawn this vmi, which will continue to fail due to the pvc issue?  otherwise the vm controller will keep scheduling a vmi pod which immediately fails.

perhaps both the vm and vmi controller shoudl share a function like the `handlePCVMisuse()` function in order to detect this situation. We probably need this at both levels of the controller

Reply:
Is there a (sensible) way to "bubble up" an error/condition/event from a VMI to the VM that spawned it?

Reply:
> Is there a (sensible) way to "bubble up" an error/condition/event from a VMI to the VM that spawned it?

yes, but i'm not sure that helps.  If we decide to not respawn the the VMI due to a specific type of error that bubbled up to the VM, then it's unclear when it's safe to respawn the failed VMI again. 

So, yes, it's possible to bubble up the information to the VM, but actually having that information influence the VMs behavior is much trickier. I wouldn't go down that path.

This issue of PVC misuse impacts both VMIs and VMs.. VMIs need to fail if someone uses the VMI directly with incorrect PVC usage, and VMs shouldn't start the VMI when incorrect usage is detected.

Typically this is all supposed to be handled at the validation webhook during creation, however it's possible (and valid) to post a VMI before the PVC exists. The VMI will eventually get spawned once the PVC is created, so we can't depend on the validation webhook catching this during create. That's what's making this a difficult issue.